kolibri/plugins/hooks.py

```diff
@@ -203,6 +203,19 @@ def inner(instance, *args, **kwargs):
     return inner
 
 
+def only_one_registered(func):
+    """
+    Ensures that only one hook of this type is registered at a time
+    """
+
+    @functools.wraps(func)
+    def inner(instance, *args, **kwargs):
+        assert len(instance.registered_hooks) == 1
```

Comment:
As above, need to coerce to something that has a length.

If taking the len of a list is unappealing can do something like:
`sum(1 for i in instance.registered_hooks)` to get the length.